htdocs/core/class/html.formmail.class.php

```diff
@@ -1096,7 +1097,7 @@ public function get_form($addfileaction = 'addfile', $removefileaction = 'remove
 	 */
 	public function getHtmlForTo()
 	{
-		global $langs, $form;
+		global $langs, $form, $object;
```

Comment:
Using $object as global variable has unpredicable results. ($object is often used by different purposes).
Also later having the selection of $withtoselected done inside the method breaks the idea the component html.formmail.class.

So I suggest to add a proterty ->withtoselected into the component html.formmail.class that will be used to set $withtoselected if defined. 
So the content of the preselected email can be set by the caller by setting ->withtoselected.
The code added into this method to set $withtoselected can then be moved into the caller to fill ->withtoselected.

Reply:
ok, i use global to be as close as possible with other dolibarr code, but that's a good news to avoid global object :)